typescript/sdk/src/worker.ts

```diff
@@ -0,0 +1,216 @@
+    // Additional models for which we want to create tables in the DB.
+    models: typeof Base[];
+  }) {
+    const options = pgConnParse(config.db_connection_uri);
+    const port = options.port || "5432";
+
+    if (!options.host || !options.database) {
+      throw new Error(
+        "Invalid postgres connection string. e.g. postgres://someuser:somepassword@somehost:5432/somedatabase",
+      );
+    }
+
+    // Build the DB connection object.
+    const dataSource = createDataSource({
+      host: options.host!,
+      port: Number(port),
+      username: options.user,
+      password: options.password,
+      database: options.database,
+      enableSSL: options.ssl as boolean,
+      additionalEntities: models,
+    });
+
+    this.config = config;
+    this.processor = processor;
+    this.dataSource = dataSource;
+  }
+
+  // This function is the main entry point for the processor. It will connect to the txn
+  // stream and start processing transactions. It will call the provided parse function to
+  // parse transactions and insert the parsed data into the database.
+  async run({ perf }: { perf?: number } = {}) {
+    await this.dataSource.initialize();
+
+    // Create the grpc client.
+    const client = new aptos.indexer.v1.RawDataClient(
+      this.config.grpc_data_stream_endpoint,
+      ChannelCredentials.createSsl(),
+      {
+        "grpc.keepalive_time_ms": 1000,
+        // 0 - No compression
+        // 1 - Compress with DEFLATE algorithm
+        // 2 - Compress with GZIP algorithm
+        // 3 - Stream compression with GZIP algorithm
+        "grpc.default_compression_algorithm": 2,
+        // 0 - No compression
+        // 1 - Low compression level
+        // 2 - Medium compression level
+        // 3 - High compression level
+        "grpc.default_compression_level": 3,
+        // -1 means unlimited
+        "grpc.max_receive_message_length": -1,
+        // -1 means unlimited
+        "grpc.max_send_message_length": -1,
+      },
+    );
+
+    const startingVersion = BigInt(this.config.starting_version || 0n);
+
+    console.log(`Requesting stream starting from version ${startingVersion}`);
+
+    const request: aptos.indexer.v1.GetTransactionsRequest = {
+      startingVersion,
+    };
+
+    const metadata = new Metadata();
+    metadata.set(
+      "Authorization",
+      `Bearer ${this.config.grpc_data_stream_api_key}`,
+    );
+
+    // Create and start the streaming RPC.
+    let currentTxnVersion = startingVersion;
+    const stream = client.getTransactions(request, metadata);
+
+    const timer = new Timer();
+    timer.start();
+
+    stream.on(
+      "data",
+      async (response: aptos.indexer.v1.TransactionsResponse) => {
+        stream.pause();
+        const transactions = response.transactions;
+
+        if (transactions == null) {
+          return;
+        }
+
+        // Validate response chain ID matches expected chain ID
+        if (response.chainId != this.config.chain_id) {
```

Comment:
Same comment as above, we need to change this to compare it to the `chain_id` stored in `ledger_info`.